lib/bn.js

```diff
@@ -494,6 +494,124 @@ BN.prototype.neg = function neg() {
   return r;
 };
 
+
+// Or `num` with `this` in-place
+BN.prototype.ior = function ior(num) {
+  this.sign = this.sign || num.sign;
+
+  while (this.length < num.length) this.words[this.length++] = 0;
+
+  for (var i = 0; i < num.length; i++) {
+    this.words[i] = this.words[i] | num.words[i];
+  }
+
+  return this._normSign();
+};
+
+
+// Or `num` with `this`
+BN.prototype.or = function or(num) {
+  if (this.length > num.length)
+    return this.clone().ior(num);
+  else
+    return num.clone().ior(this);
+};
+
+
+// And `num` with `this` in-place
+BN.prototype.iand = function iand(num) {
+  this.sign = this.sign || num.sign;
+
+  // a.length > b.length
+  var a;
+  var b;
+  if (this.length > num.length) {
+    a = this;
+    b = num;
+  } else {
+    a = num;
+    b = this;
+  }
+  for (var i = 0; i < b.length; i++) {
+    this.words[i] = this.words[i] & num.words[i];
+  }
+
+  for (; i < a.length; i++) {
```

Comment:
Actually there is no point in this, please just set the length below as you do and remove the loop.